tests/test_decision_maker/test_base.py

```diff
@@ -66,6 +67,23 @@ def test_initialisation(self):
         assert self.ownership_state.quantities_by_good_id is not None
         assert self.ownership_state.is_initialized
 
+    def test_body(self):
+        """Test the setter for the body."""
+        msg = InternalMessage()
+        msg.body = {"test_key": "test_value"}
+
+        other_msg = InternalMessage(body={"test_key": "test_value"})
+        assert msg == other_msg, "Messages should be equal."
+        assert msg.check_consistency(), "It is true."
+        assert str(msg) == msg.__str__()
```

Comment:
`str(obj)` does call `msg.__str__()`. So the string function is not being really tested.

Ideally, we should have something like:
```python
assert str(msg) == "the string representation of the message that you expect"
```